codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ConstrainedShapeSymbolProviderTest.kt

```diff
@@ -64,24 +73,31 @@ class ConstrainedShapeSymbolProviderTest {
     private val symbolProvider = serverTestSymbolProvider(model, serviceShape)
     private val constrainedShapeSymbolProvider = ConstrainedShapeSymbolProvider(symbolProvider, model, serviceShape)
 
-    private val constrainedMapShape = model.lookup<MapShape>("test#ConstrainedMap")
-    private val constrainedMapType = constrainedShapeSymbolProvider.toSymbol(constrainedMapShape).rustType()
+    @ParameterizedTest
+    @MethodSource("getShapes")
+    fun `it should return a constrained type for a constrained shape`(shapeName: String, shapeCheck: (Shape) -> Boolean) {
+        val constrainedShape = model.lookup<Shape>("test#$shapeName")
+        assert(shapeCheck(constrainedShape))
+        val constrainedType = constrainedShapeSymbolProvider.toSymbol(constrainedShape).rustType()
 
-    @Test
-    fun `it should return a constrained string type for a constrained string shape`() {
-        val constrainedStringShape = model.lookup<StringShape>("test#ConstrainedString")
-        val constrainedStringType = constrainedShapeSymbolProvider.toSymbol(constrainedStringShape).rustType()
-
-        constrainedStringType shouldBe RustType.Opaque("ConstrainedString", "crate::model")
+        constrainedType shouldBe RustType.Opaque(shapeName, "crate::model")
     }
 
-    @Test
-    fun `it should return a constrained map type for a constrained map shape`() {
-        constrainedMapType shouldBe RustType.Opaque("ConstrainedMap", "crate::model")
+    companion object {
+        @JvmStatic
+        fun getShapes(): Stream<Arguments> {
+            return Stream.of(
+                Arguments.of("ConstrainedInteger", { s: Shape -> s.isIntegerShape }),
+                Arguments.of("ConstrainedString", { s: Shape -> s.isStringShape }),
+                Arguments.of("ConstrainedMap", { s: Shape -> s.isMapShape }),
+            )
+        }
```

Comment:
If your function body comprises a single expression that is returned, you can use this shorthand way of writing functions.

```suggestion
        fun getShapes(): Stream<Arguments> =
            Stream.of(
                Arguments.of("ConstrainedInteger", { s: Shape -> s is IntegerShape }),
                Arguments.of("ConstrainedString", { s: Shape -> s is StringShape }),
                Arguments.of("ConstrainedMap", { s: Shape -> s is MapShape }),
            )
```

Doesn't really matter in this case, but we prefer the instanceof checks (e.g. `s is StringShape`) over the Smithy library-provided method calls, because the Kotlin compiler can do type narrowing in subsequent code.

Can you also move this function before the first test?

Reply:
Thanks for the explanation!
Done in https://github.com/awslabs/smithy-rs/pull/2005/commits/0ec9f136a9ff170ed0f3ca4c0a87e0ce77fd90ec